core/plugins/org.polarsys.capella.core.ui.search_cdb/src/org/polarsys/capella/core/ui/search_cdb/CapellaSearchResultTreeContentProvider.java

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.ui.search_cdb;
+
+import java.util.Set;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.jface.viewers.ITreeContentProvider;
+import org.eclipse.jface.viewers.Viewer;
+import org.polarsys.capella.core.platform.sirius.ui.navigator.viewer.CapellaNavigatorContentProvider;
+
+public class CapellaSearchResultTreeContentProvider extends CapellaNavigatorContentProvider {
+  private CapellaSearchResult_CDB searchResult;
+
+  public CapellaSearchResultTreeContentProvider(CapellaSearchResultPage_CDB capellaSearchResultPage) {
+    super();
+    searchResult = capellaSearchResultPage.getInput();
+  }
+  
+  public Object[] getChildren(Object parentElement) {
+    if (parentElement instanceof CapellaSearchMatch_CDB) {
+      return ((CapellaSearchMatch_CDB)parentElement).getMatchOccurrences().toArray();
+    }
+    return super.getChildren(parentElement);
+  }
+  
+  public Object getParent(Object element) {
+    if (element instanceof CapellaSearchMatch_CDB) {
+      return super.getParent(((CapellaSearchMatch_CDB)element).getElement());
+    }
+    return super.getParent(element);
+  }
+  
+  @Override
+  public Object[] getElements(Object inputElement) {
+    if (inputElement instanceof CapellaSearchResult_CDB) {
+      CapellaSearchResult_CDB input = (CapellaSearchResult_CDB) inputElement;
+      Set<IProject> files = input.getProjects();
+      return files.toArray();
+    }
+    return super.getElements(inputElement);
+  }
+  
+  @Override
+  public boolean hasChildren(Object element) {
+    if (element instanceof IProject) {
+      return true;
+    } else if (element instanceof CapellaSearchMatch_CDB) {
+      return !((CapellaSearchMatch_CDB) element).getMatchOccurrences().isEmpty();
+    }
+    return super.hasChildren(element);
+  }
+  
+  /*public Object[] getSources(Object[] input) {

```

Comment:
Will this commented code be used in the future ?

Reply:
No, I accidentally commited this file. I will remove it.